doc/crawler.rst

```diff
@@ -0,0 +1,91 @@
+Crawler -- mm2_crawler
+======================
+
+Overview
+--------
+
+The crawler scans all available mirrors if their content is up to date. The
+crawler only scans mirrors which are not disabled (host.user_active,
+host.admin_active, site.user_active, site.admin_active).
+
+The crawler retrieves the list of active mirrors from the database and
+spawns a thread for each host. The crawling of a host consists of a loop
+over all categories which have been configured for this host and one category
+is crawled after another.
+
+The crawler has the possibility to crawl a category via HTTP, FTP or RSYNC.
+For each category the protocol to crawl is selected from the available
+category protocols. RSYNC has the highest priority and is followed by HTTP
+and the protocol used last (if no other protocol is available) is FTP.
+
+After the crawl has finished (successful or unsuccessful) the duration
+and the timestamp of the crawl is recorded in the host. After the crawl of
+each category the files which are up to date in the category are stored
+in the database. Crawl failures (timeouts, network problems) usually result
+in a complete 'marked-as-not-up-to-date' of that mirror. Multiple consecutive
+crawl failures (default 4) will disable the host completely (host.user_active).
+
+The crawler requires enormous amounts of memory and for 40 threads crawling
+mirrors in parallel at least 32GB of memory are required. At the end of
+each crawl thread the garbage collector is explicitly called in the hope
+that some unused memory is freed again.
```

Comment:
We may want to specify the amount of hosts/files crawled as I guess another system (CentOS RPMFusion...) will have different numbers/needs there